### PR TITLE
Fix QP remapping middleware not recursing into joins or source queries of source queries

### LINKED ISSUE #15578: Remapped columns failure in joined subquestions
**Describe the bug**
If the subquestion contains remapped columns (e.g. in Admin/Data_Model a column has setting: ```Display values``` = ```Foreing key > Key Name```) then thequery builder fails to generate correct SQL query. The result is an error:
```
Column "Question 169.TITLE" not found; SQL statement: -- Metabase:: userID: 66 queryType: MBQL queryHash: 874c5fca927a1a76801d458331f5f26faacebb5920ea3307886aa66ebb21d103 SELECT "PUBLIC"."PRODUCTS"."CATEGORY" AS "CATEGORY", "PUBLIC"."PRODUCTS"."TITLE" AS "TITLE", "Question 169"."TITLE" AS "Question 169__TITLE", "Question 169"."sum" AS "sum" FROM "PUBLIC"."PRODUCTS" LEFT JOIN (SELECT "PUBLIC"."ORDERS"."PRODUCT_ID" AS "PRODUCT_ID", sum("PUBLIC"."ORDERS"."QUANTITY") AS "sum" FROM "PUBLIC"."ORDERS" WHERE "PUBLIC"."ORDERS"."DISCOUNT" > 0 GROUP BY "PUBLIC"."ORDERS"."PRODUCT_ID") "Question 169" ON "PUBLIC"."PRODUCTS"."ID" = "Question 169"."PRODUCT_ID" LIMIT 2000 [42122-197]
```

**To Reproduce**
1. Remap PRODUCT_ID column: In Admin/Data_Model choose Orders table and PRODUCT_ID column. Configure ```Display values``` = ```Use foreign key > TITLE```.
2. Create a new Custom question on Orders table. Summarize it: *Sum of QUANTITY*. Aggregate by *POLICY_ID*.
3. Save the question.
4. Create another new Custom question on Products table. Leave only 2 columns: CATEGORY, TITLE.
5. Join the saved question from step 3. Join on PRODUCTS.ID=ORDER.PRODUCT_ID .
6. You will find the first oddity: which ORDER.PRODUCT_ID should we choose? It is there twice (see on screenshot 1). However in this situation we can distinguish between them (based on their icons) but in our DWH database we can't. ...because our ID's and titles are both varchars.
7. The second oddity in the join is: the PRODUCT table has no PRODUCT_ID column. It should be TITLE column (see on screenshot 2).
8. Choose which columns should be visible from the joined columns. The remapped column is there twice again. This is OK (see on screenshot 3). Choose the *sum* and the the first *PRODUCT_ID* column. I suppose it is the remapped value. If not then try the second one.
9. Hit the *Visualize* button. You will se the error.

**Expected behavior**
For step 6: The remapped columns should not be visible 2 times in the join builder. Remapped columns should be listed only once and it should not use the remapped value in the join.
For step 7: Show the correct column label for the PRODUCTS table: *TITLE*.
For step 8: It would be great to distinguish the original value from the remapped. Maybe this way: the first PRODUCT_ID column will be labeled as PRODUCT_ID and the second as ```PRODUCTS -> TITLE```  (e.g. table -> column).
For step 9: A report with 4 columns should be visible: CATEGORY, TITLE, PRODUCT_ID -> TITLE, sum.

**Screenshots**
Screenshot1
Metabase_incorrect_column_label_in_subquestion

Screenshot2
Metabase_incorrect_column_label_in_PRODUCTS

Screenshot3
Metabase_join_subquestion_columns_selection

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_262-b10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_262",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.262-b10",
    "os.name": "Linux",
    "os.version": "4.15.0-133-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.39.0-rc1",
      "date": "2021-04-06",
      "branch": "release-x.39.x",
      "hash": "471eda1"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
These are bugs in the query builder and UI when joining subquestions. Subquestion are becoming more and more puplar because of their strength to solve more complicated questions without knowing SQL language.

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #15578

Because of all the fixes in #19901 and #20009 I was able to implement a much simpler fix this time around (compare to my first attempt at this, #19754)

### CODE REVIEW

File: src/metabase/query_processor/middleware/add_dimension_projections.clj [hunk 1]
```diff
@@ -166,47 +168,50 @@
...
+          (seq order-by) (assoc :order-by new-order-by)
+          (seq breakout) (assoc :breakout new-breakout)
+          (seq remaps)   (assoc ::remaps remaps)))
       ;; otherwise return query as-is
-      {:remaps source-query-remaps, :query query})))
+      (cond-> query
+        (seq source-query-remaps) (assoc ::remaps source-query-remaps)))))
+
+(s/defn ^:private add-fk-remaps :- QueryAndRemaps
+  "Add any Fields needed for `:external` remappings to the `:fields` clau se of the query, and update `:order-by`
```
Comment (dpsutton) [inline src/metabase/query_processor/middleware/add_dimension_projections.clj:203]: typo in `clau se`

**Review by dpsutton — APPROVED** [review]

File: src/metabase/query_processor/middleware/add_dimension_projections.clj [hunk 2]
```diff
@@ -166,47 +168,50 @@
...
+  and `breakout` clauses as needed. Returns a pair like `[external-remapping-dimensions updated-query]`."
```
[the remark this answers is not in the document]
Comment (camsaul) [reply src/metabase/query_processor/middleware/add_dimension_projections.clj:204]: I'm dumb. @howonlee pointed out the other thing in #20009 and I fixed it there but I accidentally stomped on the fix when I fixed merge conflicts last night. I'll fix it

File: src/metabase/query_processor/middleware/add_dimension_projections.clj (hunk 1 above)
Comment (camsaul) [reply src/metabase/query_processor/middleware/add_dimension_projections.clj:203]: Same thing, I fixed this in #20009 and didn't carry that fix over to this PR when I merged `master` in to this one last night. I'll fix that too

Comment (camsaul) [reply src/metabase/query_processor/middleware/add_dimension_projections.clj]: fixed

**Pull request merged**